core/tests/puppeteer-acceptance-tests/utilities/user/logged-out-user.ts

```diff
@@ -532,16 +532,15 @@ export class LoggedOutUser extends BaseUser {
     expectedDestinationPageUrl: string,
     expectedDestinationPageName: string
   ): Promise<void> {
-    await Promise.all([
-      this.page.waitForNavigation({waitUntil: ['load', 'networkidle0']}),
-      this.clickOn(button),
-    ]);
+    await this.clickAndWaitForNavigation(button);
 
-    expect(this.page.url())
-      .withContext(
+    try {
+      expect(this.page.url()).toBe(expectedDestinationPageUrl);
+    } catch (e) {
+      throw new Error(
```

Comment:
Maybe lets not catch and throw a custom error message as the expect method above would throw a  clear error message itself when the expectation fails.

Reply:
Done